File: ocaml/lib/hash.ml
(* BAION canonical JSON for OCaml — SHA-256 hex helper.
   Public standalone library. *)

(* CROSS-LINEAGE CONTRACT: identical UTF-8 input must produce identical
   hex output across all lineage libraries. SHA-256 is FIPS 180-4 (OCaml
   uses the digestif library; vendored or stdlib in other lineages). *)

(** Return SHA-256 lowercase hex string of the input bytes. *)
let sha256_hex (input : string) : string =
  let digest = Digestif.SHA256.digest_string input in
  Digestif.SHA256.to_hex digest

File: cpp/src/hash.cpp
// BAION SHA-256 for C++ — public standalone library.

#include "baion/hash.hpp"

#include "picosha2.h"

namespace baion::std_lib
{

// CROSS-LINEAGE CONTRACT: identical byte input must produce identical
// digest hex across all language implementations. SHA-256 is FIPS 180-4
// (vendored here as picosha2; native libraries elsewhere). Divergence
// breaks canonical-hash parity across implementations.
std::string sha256_hex(const std::string& bytes)
{
    return picosha2::hash256_hex_string(bytes);
}

} // namespace baion::std_lib
